### PR TITLE
Volume: Fix overflow with S24_4LE format output in xt-xcc build

### PR DESCRIPTION
**Pull request by singalsu**
The HiFi3 version of volume code got a mistake that the
rounding shift instruction can bring the value above 24 bit
range. Therefore the data needs to be shifted once again to
32 bit MSB side to saturate and then back to LSB to
have properly saturated S24_4LE (and S32_LE samples).

The patch renames the shift_right variable because it is used
now for both right and left shifts.

Signed-off-by: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>

### CODE REVIEW

File: src/audio/volume_hifi3.c [hunk 1]
```diff
@@ -271,7 +273,9 @@ static void vol_s24_to_s24_s32(struct comp_dev *dev, struct comp_buffer *sink,
 			out_sample = AE_ROUND32F48SSYM(AE_SRAI64(mult, 1));
 
 			/* Shift for S24_LE */
-			out_sample = AE_SRAA32S(out_sample, shift_right);
+			out_sample = AE_SRAA32S(out_sample, shift);
```
Comment (singalsu) [inline src/audio/volume_hifi3.c]: I think this should be AE_SRAA32RS().

Comment (singalsu) [reply src/audio/volume_hifi3.c]: Yep, a mistake. I just checked that cmocka passed with AE_SRAA32RS().

**Comment by singalsu** [comment]
I updated this PR with version that now uses rounded shift to convert to S24_4LE where it was missing.

**Comment by singalsu** [comment]
The cmocka didn't catch the lack of round because the test signal is such that it doesn't exercise rounding in every case. Also it did not test with very large and negative sample values. I will later push a cmocka volume update. It's not included here because making float reference code to output the same bits exactly as fixed is a bit tricky. My current dev version does some 1 LSB difference.
